lib/linter/linter.js

```diff
@@ -1010,6 +1016,27 @@ class Linter {
         });
 
         this.version = pkg.version;
+        this._setCwd(cwd);
```

Comment:
Would you use `internalSlotsMap` to store private data? We have experienced that people use private data directly via `_foo`-like properties. Therefore we want to hide private data in public classes.

I'd like to recommend just putting `cwd` to the internal slots instead of `_setCwd`. For example,

```diff
        internalSlotsMap.set(this, {
+           cwd: normalizeCwd(cwd),
            lastConfigArray: null,
            lastSourceCode: null,
            parserMap: new Map([["espree", espree]]),
            ruleMap: new Rules()
        });
```